src/oiiotool/expressions.cpp

```diff
@@ -317,6 +317,35 @@ Oiiotool::express_parse_atom(const string_view expr, string_view& s,
                 result = out.str();
                 if (result.size() && result.back() == '\n')
                     result.pop_back();
+
+            } else if (metadata == "IS_CONSTANT") {
+                std::vector<float> color((*img)(0, 0).nchannels());
+                if (ImageBufAlgo::isConstantColor((*img)(0, 0), color)) {
+                    result = "1";
+                } else {
+                    result = "0";
+                }
+
+
+            } else if (metadata == "IS_BLACK") {
+                auto pixstat = ImageBufAlgo::computePixelStats((*img)(0, 0));
```

Comment:
I don't think you need to call computePixelStats. You can save the expense, since you are calling isConstantColor() and that will deposit the constant color it finds into `color`, then you can just check if the values of `color[0..nchans-1]` are all 0.0f to know if it's black.